src/main/java/fr/epicanard/globalmarketchest/economy/VaultEconomy.java

```diff
@@ -123,8 +122,8 @@ public double getMoneyOfPlayer(UUID playerUUID) {
     return this.economy.getBalance(player);
   }
 
-  public void exchangeMoney(UUID playerTake, UUID playerGive, Double price) {
+  public void exchangeMoney(UUID playerTake, UUID playerGive, Double price, Double priceAfterTax) {
     this.takeMoneyToPlayer(playerTake, price);
-    this.giveMoneyToPlayer(playerGive, price);
+    this.giveMoneyToPlayer(playerGive, priceAfterTax);
```

Comment:
Isn't it the opposite ? Because you are creating money instead of destroying it.

```suggestion
    this.takeMoneyToPlayer(playerTake, priceAfterTax);
    this.giveMoneyToPlayer(playerGive, price);
```

Reply:
Yeah, like I said in the above comment, I was going with the opposite option. This way will probably require more changes to the gui menus to show the new price to the buyer